desc/objectives/linear_objectives.py

```diff
@@ -3238,3 +3227,314 @@ def compute(self, params, constants=None):
+            things=field,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        field = self.things[0]
+
+        # find indices to fix
+        if self._indices is False or self._indices is None:  # no indices to fix
+            self._idx = np.array([], dtype=int)
+        elif self._indices is True:  # all indices
+            self._idx = np.arange(np.size(self.target))
+        else:  # specified indices
+            self._idx = np.atleast_1d(self._indices)
+
+        self._dim_f = self._idx.size
+
+        self.target, self.bounds = self._parse_target_from_user(
+            self._target_from_user, field.x_lmn[self._idx], None, self._idx
+        )
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute fixed omnigenity error.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of field degrees of freedom, eg OmnigenousField.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        f : ndarray
+            Fixed omnigenity error.
+
+        """
+        return params["x_lmn"][self._idx]
+
+
+# TODO: check that this is consistent with new API of other linear objectives
+class StraightBmaxContour(_FixedObjective):
+    """Ensures the B_max contour is straight in Boozer coordinates.
+
+    Parameters
+    ----------
+    field : OmnigenousField, optional
+        Field that will be optimized to satisfy the Objective.
+    target : float, optional
+        Target value(s) of the objective. If None, uses field value.
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+    weight : float, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _target_arg = "x_lmn"
+    _fixed = False  # not "diagonal", since it is fixing a sum
+    _units = "(rad)"
+    _print_value_fmt = "Straight B_max error: {:10.3e} "
+
+    def __init__(
+        self,
+        field,
+        target=0,
```

Comment:
Might make more sense to not have target/bounds etc for this, like the self consistency constraints or `FixLambdaGauge` etc.

Reply:
I edited the logic to make it more similar to the other fixed objectives. But we actually don't want to always enforce this to have `target=0` like we do for `FixLambdaGauge`. Even though the B_max contour _should_ always be straight in perfect omnigenity, we also know this is not achievable in practice. So it's possible that you could actually get better results overall by allowing for finite error with this objective (such as by using `bounds`). This is also why I don't think we should add this objective to `_maybe_add_self_consistency`.